### PR TITLE
Wizard recipe: zlibng-v2.0.5

### PR DESCRIPTION
**Pull request by Moelf**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: zlibng
* Version: v2.0.5

@staticfloat please review and merge.

### CODE REVIEW

**Comment by Moelf** [comment]
https://github.com/zlib-ng/zlib-ng/blob/develop/.github/workflows/release.yml

supposedly they support more platforms

**Comment by vchuravy** [comment]
And if we want to use it as a stdlib https://github.com/JuliaLang/julia/issues/42218 we will need to support all Julia platforms.

File: Z/zlibng/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+products = [
+    LibraryProduct("libz", :libz)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (vchuravy) [inline Z/zlibng/build_tarballs.jl:41]: Need to add experimental_platforms here for `aarch64-darwin`

**Comment by Moelf** [comment]
I see some kind of `can't force architecture` error when compiling on other architecture which I don't quite understand and couldn't find via search engine either.

**Comment by Moelf** [comment]
```
[ Info: Found a valid dl path libzlib1.dll while looking for libz
[ Info: Could not locate libz inside ["/agent/_work/1/s/Z/zlibng/build/x86_64-w64-mingw32/JZMAMuRa/x86_64-w64-mingw32-libgfortran3-cxx03/destdir/bin"]
┌ Error: Built zlibng but libz still unsatisfied:
└ @ BinaryBuilder /depot/packages/BinaryBuilder/9NAaM/src/AutoBuild.jl:802
##[error]Built zlibng but libz still unsatisfied:
```
mingw32 is just confused about `dll` name I think.

**Comment by Moelf** [comment]
>I'm just saying using cmake on Cygwin with ZLIB_COMPAT set is unsupported combination due to limits of cmake.

I don't understand our building process on Windows at all, so I have to ask people here what to do :(

**Comment by Crghilardi** [comment]
I am able to get the build script working on `x86_64-mingw-*` by modifying the products array to include the mingw platform specific naming `libzlib1` like:
```
# The products that we will ensure are always built
products = [
    LibraryProduct(["libz", "libzlib1"], :libz)
]
``` 
See also the last part of [the API docs](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.LibraryProduct) for more info. 

If that changes in future versions, you can update it back to a single value or modify the array.

**Comment by Moelf** [comment]
this wouldn't be as hard of "Cannot force an architecture" returns 0 result when searched online

**Comment by Moelf** [comment]
oh, I thought it's from `make`, maybe we can add a `BinaryBuilder: ` prefix to errors thrown by us

**Comment by Crghilardi** [comment]
So looking at the Azure logs on `armv7l-linux-musleabihf`, the `-march` flag being applied is actually `-march=armv8-a+crc` so we're hitting [this instrinsic check](https://github.com/zlib-ng/zlib-ng/blob/f456924eaa9f08a3557cced8ed40b7ef6f9d76ea/cmake/detect-intrinsics.cmake#L4-L7). I am assuming the "forcing" error is related to applying arm8 stuff on arm7?

I tried it with [-DWITH_NATIVE_INSTRUCTIONS](https://github.com/zlib-ng/zlib-ng/blob/071e4da02377b05f5f4a5a1db9f07ea3507a5d9f/CMakeLists.txt#L81) but that still fails

Further down, there is the entry point for the [ARM intrinsics checks](https://github.com/zlib-ng/zlib-ng/blob/071e4da02377b05f5f4a5a1db9f07ea3507a5d9f/CMakeLists.txt#L97-L99), and I was able to get it to compile by disabling those in the cmake statement like:

```
cmake -DCMAKE_INSTALL_PREFIX=$prefix \
-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
-DCMAKE_BUILD_TYPE=Release \
-DZLIB_COMPAT=ON \
-DZLIB_ENABLE_TESTS=OFF \
-DBUILD_SHARED_LIBS=ON \
-DWITH_ACLE=OFF \
-DWITH_NEON=OFF

```

**Comment by Crghilardi** [comment]
I not sure if the `CMake` can be patched upstream to make BB happy for `arm` at least? Looking at the [GCC docs](https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html#ARM-Options), it sounds like those instructions are only available through the `-march=` flag. I can't seem to find individual flags that do the same thing?

**Comment by Moelf** [comment]
I'm not sure how other binary building scripts do this, but I meant that we patch the `.cmake` file of `zlib-ng` such that it doesn't set `-march`

**Comment by Moelf** [comment]
for posterity, try LibDeflate.jl

**Pull request closed without merging**